extensions/notebook/src/dialog/configurePython/configurePythonWizard.ts

```diff
@@ -0,0 +1,217 @@
+import { PickPackagesPage } from './pickPackagesPage';
+import { JupyterServerInstallation, PythonPkgDetails, PythonInstallSettings } from '../../jupyter/jupyterServerInstallation';
+import * as utils from '../../common/utils';
+import { promises as fs } from 'fs';
+import { Deferred } from '../../common/promise';
+import { PythonPathInfo, PythonPathLookup } from '../pythonPathLookup';
+import { ApiWrapper } from '../../common/apiWrapper';
+
+const localize = nls.loadMessageBundle();
+
+export interface ConfigurePythonModel {
+	kernelName: string;
+	pythonLocation: string;
+	useExistingPython: boolean;
+	pythonPathsPromise: Promise<PythonPathInfo[]>;
+	packagesToInstall: PythonPkgDetails[];
+	installation: JupyterServerInstallation;
+}
+
+export class ConfigurePythonWizard {
+	private readonly InstallButtonText = localize('configurePython.okButtonText', "Install");
+	public readonly InvalidLocationMsg = localize('configurePython.invalidLocationMsg', "The specified install location is invalid.");
+	private readonly PythonNotFoundMsg = localize('configurePython.pythonNotFoundMsg', "No python installation was found at the specified location.");
+
+	private _wizard: azdata.window.Wizard;
+	private model: ConfigurePythonModel;
+
+	private _setupComplete: Deferred<void>;
+	private pythonPathsPromise: Promise<PythonPathInfo[]>;
+
+	constructor(private apiWrapper: ApiWrapper, private jupyterInstallation: JupyterServerInstallation) {
+		this._setupComplete = new Deferred<void>();
+		this.pythonPathsPromise = (new PythonPathLookup()).getSuggestions();
+	}
+
+	public get wizard(): azdata.window.Wizard {
+		return this._wizard;
+	}
+
+	public get setupComplete(): Promise<void> {
+		return this._setupComplete.promise;
+	}
+
+	public async start(kernelName?: string, rejectOnCancel?: boolean, ...args: any[]): Promise<void> {
+		this.model = <ConfigurePythonModel>{
+			kernelName: kernelName,
+			pythonPathsPromise: this.pythonPathsPromise,
+			installation: this.jupyterInstallation,
+			useExistingPython: JupyterServerInstallation.getExistingPythonSetting(this.apiWrapper)
+		};
+
+		let pages: Map<number, BasePage> = new Map<number, BasePage>();
+
+		let wizardTitle: string;
+		if (kernelName) {
+			wizardTitle = localize('configurePython.wizardNameWithKernel', 'Configure Python to run {0} kernel', kernelName);
+		} else {
+			wizardTitle = localize('configurePython.wizardNameWithoutKernel', 'Configure Python to run kernels');
+		}
+		this._wizard = azdata.window.createWizard(wizardTitle);
+		let page0 = azdata.window.createWizardPage(localize('configurePython.page0Name', 'Configure Python Runtime'));
+		let page1 = azdata.window.createWizardPage(localize('configurePython.page1Name', 'Install Dependencies'));
+
+		page0.registerContent(async (view) => {
+			let configurePathPage = new ConfigurePathPage(this.apiWrapper, this, page0, this.model, view);
+			pages.set(0, configurePathPage);
+			await configurePathPage.start().then(() => {
```

Comment:
You can just await start() and then in the next line call onPageEnter. (probably want to await that as well?)